packages/components/src/internal/components/editable/models.ts

```diff
@@ -826,10 +829,13 @@ export class EditorModel
                         }
                     } else if (!(originalValue == undefined && value == undefined) && originalValue !== value) {
                         // only update if the value has changed
+                        // Issue 51967: We need to ensure that we get the processed/formatted value of the row
```

Comment:
If we wanted to tighten the change could inspect the `col.jsonType` here as well and only do this formatted processing for date values.